src/jit/unwindamd64.cpp

```diff
@@ -19,6 +19,44 @@ XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
 
 #if defined(_TARGET_AMD64_)
 
+int Compiler::mapRegNumToDwarfReg(regNumber reg)
+{
+    assert(reg >= 0 && reg <= REG_XMM15);
+    int dwarfReg = reg;
+
+    switch (reg) {
+    case REG_RAX: dwarfReg = 0; break;
+    case REG_RCX: dwarfReg = 2; break;
+    case REG_RDX: dwarfReg = 1; break;
+    case REG_RBX: dwarfReg = 3; break;
+    case REG_RSP: dwarfReg = 7; break;
+    case REG_RBP: dwarfReg = 6; break;
+    case REG_RSI: dwarfReg = 4; break;
+    case REG_RDI: dwarfReg = 5; break;
+    default:
+        if (reg >= REG_XMM0) {
+            dwarfReg++; // shift by 1 due to dwarf_reg_ip
```

Comment:
Nit: why not just use a static table of mappings? If that's not possible, then I think I'd prefer that each case simply returned rather than mutating a local and leaving the return to the end. Something like:

```
assert(reg >= 0 && reg <= REG_XMM15);

switch (reg)
{
case REG_RAX: return 0;
case REG_RCX: return 2;
case REG_RDX: return 1;
case REG_RBX: return 3;
case REG_RSP: return 7;
case REG_RBP: return 6;
case REG_RSI: return 4;
case REG_RDI: return 5;
default:
	return reg < REG_XMM0 ? reg : reg + 1; // shift by 1 due to dwarf_reg_ip
}
```